src/v2/view-builder/views/email/BaseAuthenticatorEmailView.js

```diff
@@ -34,9 +35,23 @@ const ResendView = View.extend(
     },
 
     showCalloutWithDelay() {
-      this.showMeTimeout = _.delay(() => {
-        this.$el.removeClass('hide');
-      }, SHOW_RESEND_TIMEOUT);
+      const timeStamp = sessionStorageHelper.getResendTimestamp();
+      if (!timeStamp) {
+        sessionStorageHelper.setResendTimestamp(Date.now());
+      }
+
+      // We keep track of a 'global' timestamp in sessionStorage because if the SIW does a re-render,
+      // we don't want to force the user to wait another 30s again to see the resend link. With this
+      // the user will wait AT MOST 30s until they see the resend link.  
+      const start = sessionStorageHelper.getResendTimestamp();
+      this.showMeTimeout = setInterval(() => {
```

Comment:
You probably don't want to create more than one setInterval timer?

Reply:
@sherizada-okta  can you clarify what the issue here is? I'm not sure what you're referring to